src/main/java/dev/meyi/bn/modules/CraftingModule.java

```diff
@@ -0,0 +1,147 @@
+package dev.meyi.bn.modules;
+import dev.meyi.bn.BazaarNotifier;
+import dev.meyi.bn.utilities.EnchantedCraftingHandler;
+import dev.meyi.bn.utilities.ColorUtils;
+import dev.meyi.bn.utilities.Defaults;
+import net.minecraft.client.gui.Gui;
+import net.minecraft.client.Minecraft;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+public class CraftingModule extends Module{
+
+    public CraftingModule() {
+        super();
+    }
+
+    public CraftingModule(JSONObject module) {
+        super(module);
+    }
+
+
+    Integer LongestXString;
+    @Override
+    protected void draw() {
+
+        if (BazaarNotifier.bazaarDataRaw != null) {
+            List<LinkedHashMap<String, Color>> items = new ArrayList<>();
+
+            ArrayList<ArrayList<String>> list;
+            list = EnchantedCraftingHandler.getBestEnchantRecipes();
+
+            for(int i = 0; i<BazaarNotifier.config.getInt("craftingLength"); i++) {
+                LinkedHashMap<String, Color> message = new LinkedHashMap<>();
+                try{
+
+                    Double profitInstaSell = Double.valueOf(list.get(i).get(0));
+                    Double profitSellOffer =Double.valueOf(list.get(i).get(1));
+                    Double PricePerMil =Double.valueOf(list.get(i).get(2));
+                    String ItemName = list.get(i).get(3);
+
+
+                    String ItemNameC = BazaarNotifier.bazaarConversions.getString(ItemName);
+                    message.put(String.valueOf(i + 1), Color.MAGENTA);
+                    message.put(". ", Color.MAGENTA);
+                    message.put(ItemNameC, Color.CYAN);
+                    message.put(" - ", Color.GRAY);
+                    message.put(BazaarNotifier.df.format(profitInstaSell), getColor(profitInstaSell.intValue()));
+                    message.put(" / ", Color.GRAY);
+                    message.put(BazaarNotifier.df.format(profitSellOffer), getColor(profitSellOffer.intValue()));
+                    message.put(" /  ", Color.GRAY);
+                    message.put(BazaarNotifier.df.format(PricePerMil), getColor(PricePerMil.intValue()));
+
+
+
+
+                }catch(IndexOutOfBoundsException e) {
+                    message.put("ERROR, JUST WAIT...",Color.RED);
+                }

```

Comment:
Is there a better way to do this? Perhaps like it was done in https://github.com/symt/BazaarNotifier/blob/master/src/main/java/dev/meyi/bn/modules/SuggestionModule.java#L25-L52